tensordict/memmap.py

```diff
@@ -637,7 +619,7 @@ def __setitem__(self, idx: INDEX_TYPING, value: torch.Tensor):
                 )
             self.memmap_array[idx] = to_numpy(value)
 
-    def __setstate__(self, state: Dict[str, Any]) -> None:
+    def __setstate__(self, state: dict[str, Any]) -> None:
```

Comment:
Why dict instead of Dict?

Reply:
Since Python 3.9, all Python builtin container types support `__class_getitem__`. For example, `list[int]`, `dict[str, Tensor]`, `collections.OrderedDict[str, Tensor | None]`, `collections.abc.Sequence[int]`. This behavior can be enabled by `from __future__ import annotations` in Python 3.7 since the annotations are postend evaluated. The old `typing.List`, `typing.Dict`, etc., are deprecated and may be removed after 5 years since Python 3.9. That is 2024 when Python 3.13 is released.

Reply:
See https://docs.python.org/3/library/typing.html#typing.Dict

> Deprecated since version 3.9: [builtins.dict](https://docs.python.org/3/library/stdtypes.html#dict) now supports subscripting ([]). See [PEP 585](https://peps.python.org/pep-0585/) and [Generic Alias Type](https://docs.python.org/3/library/stdtypes.html#types-genericalias).